ViewControllers/DetailViewController.swift

```diff
@@ -141,36 +152,42 @@ class DetailViewController: UIViewController {
         return [languagesSection, genreSection]
         }
 
+    
     override func viewDidLoad() {
         super.viewDidLoad()
 
         guard let id = selectedMovieId else { return }
         
         movieDataService.getMovieDetails(movieId: id) { [weak self] result in
             guard let self = self else {return}
-            
+                
             switch result {
             case .success(let details):
                 self.movieDetails = details
             case .failure(let error):
-//                Here I wanted to display an alert controller with the info about the error. Not sure if error.localizedDescription will hold anything though.
                 switch error {
-                case .errorOccurred:
-                    let ac = UIAlertController(title: "Error", message: "\(error.localizedDescription)", preferredStyle: .alert)
-                    ac.addAction(UIAlertAction(title: "OK", style: .default))
-                    self.present(ac, animated: true)
-                default:
-                    break
+                case .failedToGetResponse:
+                    self.showError(message: error.localizedDescription)
+                case .failedToDecode:
+                    self.showError(message: error.localizedDescription)
+                case .failedToGetData:
+                    self.showError(message: error.localizedDescription)
```

Comment:
Since we are not doing anything different between errors this can be simply:
```swift
self.showError(message: error.localizedDescription)
```
Without the `switch` between error cases